openapi.yaml

```diff
@@ -1,279 +1,662 @@
-          schema:
-            $ref: '#/definitions/ErrorResult'
-        '500':
-          $ref: '#/responses/500'
-parameters:
-  pageNumber:
-    name: page[number]
-    in: query
-    type: integer
-    required: false
-    description: Page number of results
-    default: 1
-    minimum: 1
-  pageSize:
-    name: page[size]
-    in: query
-    type: integer
-    required: false
-    description: Number of results to return
-    default: 25
-    maximum: 500
-    minimum: 1
-responses:
-  '400':
-     description: Bad request
-     schema:
-       $ref: '#/definitions/ErrorResult'
-  '500':
-     description: Internal server error
-     schema:
-       $ref: '#/definitions/ErrorResult'
-securityDefinitions:
-  OAuth2:
-    type: oauth2
-    flow: application
-    tokenUrl: https://api.oregonstate.edu/oauth2/token
-    scopes:
-      full: Full access to the API
-definitions:
-  Meta:
-    properties:
-      totalResults:
-        type: integer
-        description: Total number of results
-        example: 10
-      totalPages:
-        type: integer
-        description: Total number of pages
-        example: 10
-      currentPageNumber:
-        type: integer
-        description: Page number of the returned results
-        example: 1
-      currentPageSize:
-        type: integer
-        description: Number of results per page
-        example: 25
-  SelfLink:
-    properties:
-      self:
-        type: string
-        format: uri
-        description: Self-link of current resource
-  PaginationLinks:
-    properties:
-      first:
-        type: string
-        format: uri
-        description: The first page of data
-      last:
-        type: string
-        format: uri
-        description: The last page of data
-      prev:
-        type: string
-        format: uri
-        description: The previous page of data
-      next:
-        type: string
-        format: uri
-        description: The next page of data
-  PetResource:
-    properties:
-      id:
-        type: string
-        description: A unique ID of pet
-      type:
-        $ref: '#/definitions/PetType'
-      links:
-        $ref: '#/definitions/SelfLink'
-      attributes:
-        properties:
-          name:
             type: string
-            description: Name of pet
-            example: Hedwig
-          owner:
+      responses:
+        "200":
+          description: "Successful response"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/ServicesResultObject"
+        "404":
+          description: "Resource not found"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Errors"
+        "500":
+          description: "Internal Server Error"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Errors"
+components:
+  parameters:
+    pretty:
```

Comment:
I haven't seen this parameter in any newer api that returns a JSON object. This is more of a question for @tsoliangwu0130 , but do we have users who need this feature? Otherwise I think we could remove it.

Reply:
@kalcicd good point. I also left my comment regarding this in the above comment. Long story short, this parameter was used in the very first version of our old DW skeleton and JSON response was not prettified by default. However, since we have switched to the new express skeleton, I think this param is not necessary anymore.